Bottom: 5fdbc6e3702d7daa077199e1a5f045bdae499a63
Top:    ec505f3b99dea7308a81bda5d3ae0e5a43c82c8e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 14:42:53 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index f1f2ecd..6632e75 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -227,6 +227,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
